### PR TITLE
#184581182 set user roles or permissions

### PR DESCRIPTION
**Pull request by Emile-SHUMBUSHO**
#### What does this PR do?
This PR implements functionality for the admin to set roles or permissions to other users

#### Description of Task to be completed?
- a user with admin privileges should be able to set roles and permissions
- list all users
- creates roles
- creates permissions
- assign roles to the user
- assign permissions to the role

#### How should this be manually tested?
- Clone this repo git clone https://github.com/atlp-rwanda/e-comm-team-amigos-bn
- Switch your working directory cd e-comm-team-amigos-bn
- Create an .env file and update it with .env-example
- Install all the dependencies by running npm install
- Then run the project by typing npm start

#### Endpoints to be tested:
-`localhost:400/user/users`
-`localhost:400/user/id`

#### Any background context you want to provide?
- Have Node Js installed in your local machine

#### What are the relevant pivotal tracker/Trello stories?

Story ID: [#184581182](https://www.pivotaltracker.com/n/projects/2632829/stories/184581182)

#### Screenshots (if appropriate)
swager ui documentation
Screenshot 2023-03-19 at 18 09 29
passed test
Screenshot 2023-03-19 at 23 29 25

### CODE REVIEW

**Review by Mystica52 — COMMENTED** [review]
resolve the issue in your circle ci

File: src/tests/roles.test.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+chai.use(chaiHttp);
+
+describe("Set user roles or permissions", () => {
+  before(async () => await models.User.destroy({ where: {} }));
+
+  it("Should return all users", async()=>{
+    const res = await chai.request(app).get('/user/users').send();
+    expect(res).to.have.status(200);
+    expect(res.body.message).to.equal('Users retreived successfully');
+    // expect(res.body.count).to.equal(res.length);
```
Comment (nshutijonathan) [inline src/tests/roles.test.js]: remove commented out codes from your the repo

**Review by nshutijonathan — COMMENTED** [review]
Remove commented out codes

**Review by 1MuhireDavid — COMMENTED** [review]
Please remember to edit the PR description and remove the commit message.

**Review by GasanaJr — CHANGES_REQUESTED** [review]
Is it possible to attach a screenshot of passing tests in the PR body?

**Review by wilbrord2 — APPROVED** [review]

**Review by GasanaJr — APPROVED** [review]
Great Job

**Review by AbdulKhaliq59 — APPROVED** [review]

**Review by Habinezajanvier — APPROVED** [review]
👍

**Pull request merged**